solana/restaking/programs/restaking/src/lib.rs

```diff
@@ -330,6 +324,10 @@ pub mod restaking {
         if vault_params.service.is_some() {
             return Err(error!(ErrorCodes::ServiceAlreadySet));
         }
+        let token_account = &ctx.accounts.receipt_token_account;
```

Comment:
can you move this validation to the beginning of the function?

Reply:
sure sure